test/unit/TestLanguagePipeline.hs

```diff
@@ -282,6 +282,12 @@ testLanguagePipeline =
                       ]
                     )
             )
+        , testCase
+            "simple type ascription"
+            (valid "(3 : int) + 5")
+        , testCase
+            "invalid type ascription"
+            (process "1 : text" "1: Can't unify text and int")
```

Comment:
Can we add a few more tests?  e.g. let's at least have a test where we annotate something with a polytype.  Actually, trying it out just now, this doesn't seem to work:
```
((\x . x) : a -> a) 3
```
This ought to return `3`, but instead we get an error `Can't unify s0 and int`.  I think I see what the issue is, I will add a comment to the typechecker.

Reply:
I have added this test case, but I will add more test cases tomorrow.

Reply:
I have added a few additional test cases. Do you still see a gap in the test coverage?